src/main/java/org/generationcp/middleware/dao/dms/ExperimentDao.java

```diff
@@ -1081,4 +1081,44 @@ public void updateStudyExperimentGeolocationIfNecessary(final Integer studyId, f
 		}
 	}
 
+	public List<ExperimentModel> getExperimentsByParentIds(final List<Integer> parentIds) {
+		try {
+			final Criteria criteria = this.getSession().createCriteria(this.getPersistentClass());
+			criteria.add(Restrictions.in("parent.ndExperimentId", parentIds));
+
+			return criteria.list();
+
+		} catch (final HibernateException e) {
+			final String message =
+				"Error at getExperimentsByParentID=" + parentIds + "query at ExperimentDao: " + e.getMessage();
+			ExperimentDao.LOG.error(message, e);
+			throw new MiddlewareQueryException(message, e);
+		}
+	}
+
+	/**
+	 * Replace stock_id from the parent row and its children recursively
+	 * @param observationUnitIds from the parent (plot, observation) dataset
+	 * @param newEntryId
+	 */
+	public void updateEntryId(final List<Integer> observationUnitIds, final Integer newEntryId) {
+		if (observationUnitIds.isEmpty()) {
+			return;
+		}
+		try {
+			final String hqlUpdate = "update ExperimentModel e set e.stock.id = :newEntryId where e.ndExperimentId in (:observationUnitIds)";
```

Comment:
Couldn't we alter the update statement to also include the child experiments:
`where e.ndExperimentId in (:observationUnitIds) or e.parent.ndExperimentId in (:observationUnitIds)`

This way we don't have to retrieve and store child experiments in memory then proceed to update them separately.

Reply:
I did it this way considering we could have sub-sub plots